test/membrane_rtc_engine/engine_test.exs

```diff
@@ -21,276 +21,119 @@ defmodule Membrane.RTC.EngineTest do
-  describe "accepting a new peer" do
-    test "triggers peerAccepted event", %{rtc_engine: rtc_engine} do
-      peer_id = "sample_id"
+    test "is ignored for non-peers", %{rtc_engine: rtc_engine} do
+      endpoint_spec = %MessageEndpoint{rtc_engine: rtc_engine, owner: self()}
 
-      metadata = %{
-        "displayName" => "Bob"
-      }
+      Engine.add_endpoint(rtc_engine, endpoint_spec)
+      Engine.add_endpoint(rtc_engine, endpoint_spec, endpoint_id: "not-a-peer")
 
-      media_event =
-        %{
-          type: "join",
-          data: %{
-            receiveMedia: true,
-            metadata: metadata
-          }
-        }
-        |> Jason.encode!()
-
-      Engine.receive_media_event(rtc_engine, {:media_event, peer_id, media_event})
-      assert_receive {_from, {:new_peer, ^peer_id, ^metadata}}
-      Engine.accept_peer(rtc_engine, peer_id)
-      assert_receive {_from, {:rtc_media_event, ^peer_id, media_event}}, 1000
-
-      assert %{
-               "type" => "peerAccepted",
-               "data" => %{
-                 "id" => peer_id,
-                 "peersInRoom" => [],
-                 "iceTransportPolicy" => "all",
-                 "integratedTurnServers" => []
-               }
-             } ==
-               Jason.decode!(media_event)
-    end
-  end
+      Engine.message_endpoint(
+        rtc_engine,
+        "not-a-peer",
+        {:execute_actions, [notify: {:ready, "metadata"}]}
+      )
 
-  describe "denying a new peer" do
-    test "triggers peerDenied event", %{rtc_engine: rtc_engine} do
-      peer_id = "sample_id"
+      refute_receive {:new_peer, _peer}
+      refute_receive {:ready, _peers_in_room}
+    end
 
-      metadata = %{
-        "reason" => "bob smells"
+    test "reports other peers", %{rtc_engine: rtc_engine} do
+      peer1_spec = %TrackEndpoint{
+        rtc_engine: rtc_engine,
+        track: video_track("peer1", "track1", "track1-metadata", "stream1"),
+        peer_metadata: "peer1-metadata"
       }
 
-      media_event =
-        %{
-          type: "join",
-          data: %{
-            receiveMedia: true,
-            metadata: metadata
-          }
-        }
-        |> Jason.encode!()
-
-      peer = %Peer{
-        id: peer_id,
-        metadata: metadata
-      }
+      msg_endpoint_spec = %MessageEndpoint{rtc_engine: rtc_engine, owner: self()}
 
-      Engine.receive_media_event(rtc_engine, {:media_event, peer_id, media_event})
-      assert_receive %Message.NewPeer{rtc_engine: ^rtc_engine, peer: ^peer}
-      Engine.deny_peer(rtc_engine, peer_id, data: metadata)
-      assert_receive %Message.MediaEvent{rtc_engine: ^rtc_engine, to: ^peer_id, data: data}
+      Engine.add_endpoint(rtc_engine, peer1_spec, peer_id: "peer1")
+      Engine.add_endpoint(rtc_engine, msg_endpoint_spec, peer_id: "peer2")
 
-      assert %{"type" => "peerDenied", "data" => %{"reason" => "bob smells"}} ==
-               Jason.decode!(data)
-    end
-  end
+      Process.sleep(200)
```

Comment:
This shouldn't be needed. AFAIK membrane_core guarantees that after returning `spec` action you can immediately communicate with an element.